### PR TITLE
refactor: Add `BatchedTrigger`

### PR DESCRIPTION
**Pull request by robinholzi**
# Motivation

To eliminate the batching code duplication in the trigger implementations by moving it into a dedicated abstract base class.

### CODE REVIEW

File: modyn/supervisor/internal/triggers/amounttrigger.py [hunk 1]
```diff
@@ -16,8 +18,11 @@ def __init__(self, config: DataAmountTriggerConfig):
 
         assert self.data_points_for_trigger > 0, "data_points_for_trigger needs to be at least 1"
 
+    @override
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/triggers/amounttrigger.py:21]: so far we don't use this decorator in the codebase. any reason to introduce it now here?

File: modyn/supervisor/internal/triggers/datadrifttrigger.py [hunk 2]
```diff
@@ -70,7 +73,7 @@ def __init__(self, config: DataDriftTriggerConfig):
         self.model: StatefulModel | None = None
 
         self._sample_left_until_detection = (
```
Comment (MaxiBoether) [inline modyn/supervisor/internal/triggers/datadrifttrigger.py:75]: This should be set by the new parent class now?

**Review by MaxiBoether — APPROVED** [review]
Thank you, I only had minor comments, this makes reading the batched triggers much easier.

File: modyn/supervisor/internal/triggers/amounttrigger.py (hunk 1 above)
Comment (robinholzi) [reply modyn/supervisor/internal/triggers/amounttrigger.py:21]: For mypy to tell me when I forgot to update some signature of a function😅 The decorator is afaik only typing sugar, and doesn't affect runtime behaviour.

**Pull request merged**